In >=gtk+-2.11 GtkBorder uses GSlice, and using g_free here will cause
double free corruption.  Use gtk_border_free instead.

diff -Naur gtk-orig/button.cpp gtk/button.cpp
--- src/gtk-orig/button.cpp	2005-12-15 18:11:43.000000000 -0600
+++ src/gtk/button.cpp	2007-07-28 20:58:06.000000000 -0600
@@ -89,7 +89,7 @@
             right_border += default_border->right;
             top_border += default_border->top;
             bottom_border += default_border->bottom;
-            g_free( default_border );
+            gtk_border_free( default_border );
         }
 #else
         left_border = 6;
diff -Naur gtk-orig/window.cpp gtk/window.cpp
--- src/gtk-orig/window.cpp	2006-08-21 11:09:03.000000000 -0600
+++ src/gtk/window.cpp	2007-07-28 20:58:36.000000000 -0600
@@ -3273,7 +3273,7 @@
                 right_border += default_border->right;
                 top_border += default_border->top;
                 bottom_border += default_border->bottom;
-                g_free( default_border );
+                gtk_border_free( default_border );
             }
 #else
             left_border = 6;
